server/backend/api/app/routers/router_guards.py

```diff
@@ -50,38 +41,63 @@ async def get_current_requester(
     request_cognito_user_id: str = Depends(get_request_cognito_user_id),
     access_roles: List[str] = Depends(get_access_roles),
     db: Session = Depends(database.get_db),
-):
-    LOGGER.debug(
-        f"Debug router_guard get_current_requester, current request_cognito_user_id: {request_cognito_user_id}"
-    )
+) -> Requester:
     LOGGER.debug(
-        f"Debug router_guard get_current_requester, current db connection: {db}"
+        f"Retrieving current requester from: request_cognito_user_id: {request_cognito_user_id}"
     )
-    fam_user: FamUser = crud_user.get_user_by_cognito_user_id(
+    fam_user: FamUser = crud_user.fetch_initial_requester_info(
         db, request_cognito_user_id
     )
```

Comment:
The 'fetch_initial_requester_info' method uses programmatic SqlAlehcmy ORM joined to get other information (delegated_admin, T&C) for this user.